plugins/coverbrowser.koplugin/bookinfomanager.lua

```diff
@@ -423,7 +477,27 @@ function BookInfoManager:extractBookInfo(filepath, cover_specs)
             end
             if props.title and props.title ~= "" then dbrow.title = props.title end
             if props.authors and props.authors ~= "" then dbrow.authors = props.authors end
-            if props.series and props.series ~= "" then dbrow.series = props.series end
+            if props.series and props.series ~= "" then
+                -- NOTE: If there's a series index in there, split it off to series_index, and only store the name in series.
+                --       This property is currently only set by:
+                --         * DjVu, for which I couldn't find a real standard for metadata fields
+                --           (we currently use Series for this field, c.f., https://exiftool.org/TagNames/DjVu.html).
+                --         * CRe, which could offer us a split getSeriesName & getSeriesNumber...
+                --           except getSeriesNumber does an atoi, so it'd murder decimal values.
+                --           So, instead, parse how it formats the whole thing as a string ;).
+                if string.find(props.series, "#") then
+                    dbrow.series, dbrow.series_index = props.series:match("(.*) #(.*)")
+                    if dbrow.series_index then
+                        -- We're inserting via a bind method, so make sure we feed it a Lua number, because it's a REAL in the db.
+                        dbrow.series_index = tonumber(dbrow.series_index)
```

Comment:
Is it worth it being stored as a REAL (storing `4.0` and `4`, we would be fetching the same value for both ?)  when in the end, you're just using the provided value as-is, as a string?

Reply:
The reasoning was the potential pie-in-the-sky query UI, where you would most definitely want it kept as a real to do stuff like "get me all books in series n and order them by index" ;).

(Also, I like being anal about data types in general ^^).

Reply:
Given than native Lua numbers are already all doubles, it's not really much of an issue?

Reply:
Double-checked, and fun (and welcome) fact: a round integer is displayed without a decimal by Lua.

e.g., Calibre always uses a real for series_index, so given

```
    <meta property="belongs-to-collection" id="id-3">Tinfoil Dossier</meta>
    <meta refines="#id-3" property="collection-type">series</meta>
    <meta refines="#id-3" property="group-position">3.0</meta>
```

CRe honors that, and we get `["series"] = "Tinfoil Dossier #3.0"` in `doc_props`.

BookInfo honors that, and we get `Tinfoil Dossier|3.0` in the DB.

And, then, Lua happens... and, wonders of wonders:

```
> print(tonumber(3.0))
3
```

\o/

Reply:
Which means, now I understand what the `gsub` I zapped were for, and I need to restore them for the pure doc_props case ;).

Reply:
> it's not really much of an issue?

It's not because SQLite is friendly (data type not enforced on the column, but stored with the data), and won't fail if you store `"zzz"` in a REAL column.
I think we had that discussion in the past, possibly already about the series stuff (or some other calibre metadata): you trust the data type (so, the user or publisher to have done the right thing) - while I think we can meet garbage in the wild, and that it's safer to keep the garbage by just storing everything as text - and do only the conversion at the end when it matters.

(But never really seeing any series with the kind of book I read, I don't mind much how you deal with that :)

Reply:
Or may be, split by # and only if the right part is a number, store it in series_index.
If it's not a number, let it be as part of series, in case you get "The Avengers vs. #twitter" as a serie :)

Reply:
Oh, `tonumber("foo")` returns nil, which I'm guessing will be bound as a NULL, so I assume this all shakes out all right in practice :).

Reply:
The only real issue would be roman numerals, but I don't *think* I've ever seen those in the wild ^^.

Reply:
> Oh, `tonumber("foo")` returns `nil`, which I'm guessing will be bound as a NULL, so I assume this all shakes out all right in practice :).

Oh, except for not restoring that potential bit of string to `series`, right ;).

Reply:
b7339144 ought to take care of that, thanks!

Reply:
If you're confident with that, ok :)
(But I would personally store the raw stuff in the db, and do the series parsing at display/search time - so if you get some case when the logic needs a tweak, you don't have to trash the whole DB to reindex/store something right.)